Add documentation articles for Chart and StockChart No Data Template (#2461)
telerik/blazor-docs: components/chart/templates.md

---
title: No Data Template
page_title: Chart - No Data Template

> description: The NoDataTemplate in the Chart for Blazor lets you customize the content displayed when no data is available for any series.

Reviewer: ```suggestion
description: The NoDataTemplate in the Chart for Blazor lets you customize the content displayed when no data is available for all series.
```

slug: chart-no-data-template
tags: telerik,blazor,chart,templates
published: True
position: 101
---

# No Data Template

> The `NoDataTemplate` allows you to define custom content when any of the Chart series has no data to show. To change the default **No data** localizable text, declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag:

Reviewer 2: `any` is used in negative statements or questions. Right now, this falls in neither category and `any` sounds unnatural in this instance. Try something like this.

```suggestion
When a Chart series has no data to show, you can use the `NoDataTemplate` to display custom content. This is a global Chart setting and applies to all instances where a Chart series does not have any data to show.

To change the default **No data** localizable text, declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag:
```

Adapt these changes to the identical page for the Stock chart as well.

Reviewer 2: Also, please note what's the default behavior somewhere.

Author: > Also, please note what's the default behavior somewhere.

The default behavior is mentioned in the next sentence. Is is necessary to note it somewhere else?

> When a Chart series has no data to show, you can use the `NoDataTemplate` to display custom content. This is a global Chart setting and applies to all instances where a Chart series does not have any data to show. To change the default **No data** localizable text, declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag:

Reviewer: ```suggestion
When all Chart series have no data to show, you can use the `NoDataTemplate` to display custom content. This is a global Chart setting and applies to all instances. To change the default **No data** localizable text, declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag:
```

> Starting in **version 7.0.0**, when all Chart series have no data to show, a default **No data** localizable text is displayed. You can customize this content using the `NoDataTemplate`. This is a global Chart setting and applies to all instances where a Chart series does not have any data to dispay. To change the default behavior, simply declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag:

Reviewer: ```suggestion
Starting in **version 7.0.0**, when all Chart series have no data to show, a default layout with **No data** text is displayed over the Chart. To customize the default layout content declare a `<NoDataTemplate>` tag inside a `<ChartSettings>` tag. The content in the `<NoDataTemplate>` tag is a global Chart setting and applies to all instances.
```

Reviewer: If you choose to decline the suggested change, just make sure that the instructions are not misleading in the part that the `<NoDataTemplate>` applies only to instances where a Chart series does not have any data to display. I get that from:
"This is a global Chart setting and applies to _all instances **where**_ a Chart series does not have any data to dispay."
If I understand correctly, the `<NoDataTemplate>`  applies only when **all** Chart series does not have any data.
